### PR TITLE
refactor: timestamp remapping

### PR DESCRIPTION
**Pull request by Orland0x**
remapped timestamps are now cached so only one call to the remapper needs to be made. This call should occur after the voting period has begun but before the first vote is cast.

### CODE REVIEW

File: starknet/src/utils/single_slot_proof.cairo [hunk 1]
```diff
@@ -11,7 +11,51 @@ mod SingleSlotProof {
...
+        /// Queries the Timestamp Remapper contract for the closest L1 block number that occured before
+        /// the given timestamp and then caches the result. If the queried timestamp is less than the earliest
+        /// timestamp or larger than the latest timestamp in the mapper then the transaction will revert.
+        /// This function should be used to cache a remapped timestamp before its used when calling the 
+        /// `get_storage_slot` function with the same timestamp.
+        ///
+        /// # Arguments
+        ///
+        /// * `timestamp` - The timestamp at which to query.
+        /// * `tree` - The tree proof required query the remapper.
```
Comment (pscott) [inline starknet/src/utils/single_slot_proof.cairo]:
```suggestion
        /// * `tree` - The tree proof required to query the remapper.
```

File: starknet/src/utils/single_slot_proof.cairo [hunk 2]
```diff
@@ -11,7 +11,51 @@ mod SingleSlotProof {
     #[storage]
     struct Storage {
         _timestamp_remappers: ContractAddress,
```
Comment (pscott) [inline starknet/src/utils/single_slot_proof.cairo:13]: do you know why it's `remapperS` if it's a single contract? is it a contract consisting of multiple contracts?

**Review by pscott — APPROVED** [review]
utACK

**Comment by Orland0x** [comment]
will actually wait for some more updates from herodotus before merging this (if possible for audit fixes deadline)

File: starknet/src/utils/single_slot_proof.cairo (hunk 2 above)
Comment (Orland0x) [reply starknet/src/utils/single_slot_proof.cairo:13]: there can technical me multiple mappings in the contract i think, not sure when that would be useful though.

**Pull request merged**